CommonController.java

```diff
@@ -0,0 +1,443 @@
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+		Map<String, String> params = new HashMap<String, String>();
+		params.put(Constants.MSG_SYSTEM, msg);
+		params.put("updated", "Y");
+
+		return goPost(response, "/common/editNickForm", params);
+	}
+
+	@RequestMapping("/download")
+	public String download(
+			@RequestParam HashMap<String, Object> param,
+			ModelMap model,
+			HttpServletRequest request,
+			HttpServletResponse response) {
+
+		String fileName = request.getParameter("fileName");
+		String sysFilePath = request.getParameter("sysFilePath");
+		String absolutePath = Constants.IMG_PATH + sysFilePath;
+		try {
+//			boolean isLogin = CookieUtil.isLogin(request);
+//			if(!isLogin){
+//				PrintWriter out = response.getWriter();
+//				out.println("<script>alert('You need to sign-in'); history.back();</script>");
+//				return;
+//			}
+
+			File tempFile = new File( absolutePath );
+			int filesize = (int) tempFile.length();
+			String filetype = "application/x-zip-compressed";
+			String agentType = request.getHeader("Accept-Encoding");
+			try {
+				if (!tempFile.exists() || !tempFile.canRead()) {
+					PrintWriter out = response.getWriter();
+					out.println("<script>alert('File Not Found'); history.back();</script>");
+					return null;
+				}
+			} catch (Exception e) {
+				PrintWriter out = response.getWriter();
+				out.println(
+					"<script>alert('File Not Found');history.back();</script>");
+				return null;
+			}
+
+			boolean flag = false;
+			if (agentType != null && agentType.indexOf("gzip") >= 0)
+				flag = true;
+
+			flag = false;
+
+			if (flag) {
+				response.setHeader( "Content-Encoding", "gzip" );
+				response.setHeader( "Content-disposition", "attachment;filename=" + CommonUtil.kscToasc(fileName));
+				ServletOutputStream servletoutputstream = response.getOutputStream();
+				GZIPOutputStream gzipoutputstream = new GZIPOutputStream(servletoutputstream);
+				dumpFile(tempFile, gzipoutputstream);
+
+				gzipoutputstream.close();
+				servletoutputstream.close();
+			}
+			else {
+				response.setContentType( "application/octet-stream") ;
+				response.setHeader( "Content-disposition", "attachment;filename=" + CommonUtil.kscToasc(fileName));
+				ServletOutputStream servletoutputstream1 = response.getOutputStream();
+				dumpFile(tempFile, servletoutputstream1);
+
+				servletoutputstream1.flush();
+				servletoutputstream1.close();
+			}
+
+		} catch (IOException e) {
+			PrintWriter out = null;
+			try {
+				out = response.getWriter();
+			} catch (IOException e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
+			}
+			out.println("<script>alert('File Not Found');history.back();</script>");
+			return null;
+		}
+		return null;
+	}
+
+	private void dumpFile(File realFile, OutputStream outputstream) {
+		byte readByte[] = new byte[4096];
+		try {
+			BufferedInputStream bufferedinputstream =

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *bufferedinputstream*. The resource is referenced in statements at the following lines: 274, 276. The resource closure statement is at line: 280. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by OutputStream.write. Either a) close *bufferedinputstream* in a try-finally block or b) close the resource by declaring *bufferedinputstream* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).